[Feat]-Add JSON Schema Editor tool

MadhavDhatrak opened the pull request:
<!-- In order to keep off topic discussion to a minimum, it helps if the "work to be done" is already agreed on. -->
<!-- Ideally, a GitHub Issue with the label `Status: Consensus` will have been concluded, and linked to in this PR. -->
<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
Feature: Added a new tool – JSON Schema Editor.

<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**
-  Closes #1118 


**Screenshots/videos:**

https://github.com/user-attachments/assets/dbcf75b7-f24b-4cfc-9b50-9ef80d0100f6



**If relevant, did you update the documentation?**
Documentation not required as this is a  tool addition.

**Summary**
This PR introduces the JSON Schema Editor tool to the platform. It allows users to:

- Easily build JSON schemas using an intuitive UI.
- Convert JSON schemas into formats like SQL, GraphQL, and Protocol Buffers.
- Generate sample data conforming to the defined JSON schema.
- The addition enhances developer productivity by streamlining JSON schema creation and format conversions.

**Does this PR introduce a breaking change?**

No breaking changes introduced.

DhairyaMajmudar reviewed (approved):
Thanks for the PR @MadhavDhatrak, LGTM 👍🏻

Relequestual reviewed (changes requested):
As far as I can tell, this only supports top level, and is very limited feature wise. I feel like we need to add something about this in the tooling listing notes.

MadhavDhatrak commented:
Hi @Relequestual ,

Thanks for the feedback! I've updated the PR to mention that the tool currently supports only top-level JSON schema creation.

I have also added the following tooling types to reflect the current functionality:
```'editor'```
```'util-format-conversion'```
```'util-schema-to-schema'```

Screenshot
![Screenshot 2024-12-27 163516](https://github.com/user-attachments/assets/58eba589-85b0-4b2a-af23-5918eb2183ff)

Please review my changes and let me know if any further adjustments are needed.

robclarabase commented:
> As far as I can tell, this only supports top level, and is very limited feature wise. I feel like we need to add something about this in the tooling listing notes.

Hi @Relequestual - I worked on this project and would like to make the improvements you mentioned here in a subsequent iteration (not that it should be a blocker to this PR). Just so I understand correctly, does "only supports top level" refer to the lack of support for nested properties with an "object" type? Also what features/functionality would it need for it not be considered limited? Thanks!
